identity-actor/src/actor/errors.rs

```diff
@@ -0,0 +1,121 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use libp2p::request_response::OutboundFailure;
+
+use crate::didcomm::ThreadId;
+
+/// The `Result` type for the actor.
+pub type Result<T> = std::result::Result<T, Error>;
+
+/// Errors that can occur during actor execution.
+#[non_exhaustive]
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+  #[non_exhaustive]
+  #[error("transport error during {0}")]
+  TransportError(&'static str, #[source] libp2p::TransportError<std::io::Error>),
+  #[error("invalid endpoint")]
+  InvalidEndpoint,
+  #[non_exhaustive]
+  #[error("failure during sending an outbound request and receiving the response")]
+  OutboundFailure(#[source] OutboundFailure),
+  #[error("unexpected request `{0}`")]
+  UnexpectedRequest(String),
+  #[error("handler invocation error: {0}")]
+  HandlerInvocationError(String),
+  #[error("hook invocation error: {0}")]
+  HookInvocationError(String),
```

Comment:
Hooks are no longer part of this PR per the new description so perhaps this variant can be removed for now?